pkg/cloud/services/networking/client.go

```diff
@@ -0,0 +1,327 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networking
+
+//go:generate mockgen -destination=mock_client/client.go -package=mock_client sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/services/networking NetworkClient
+//go:generate /usr/bin/env bash -c "cat ../../../../hack/boilerplate/boilerplate.generatego.txt mock_client/client.go > mock_client/_client.go && mv mock_client/_client.go mock_client/client.go"
```

Comment:
CAPA and CAPZ write these two line in mock_xx/doc.go. 

- https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/ec2/mock_ec2iface/doc.go
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/azure/services/subnets/mock_subnets/doc.go

Let's follow them which means creating new file pkg/cloud/services/networking/mock_networking/doc.go and write two lines into it.

Reply:
Thanks! I agree with keeping this the same as CAPA and CAPZ. I've updated the package name and moved the `//go:generate` to doc.go. I've also squashed commits.